cpp/src/fil/infer.cu

```diff
@@ -345,11 +346,14 @@ struct tree_aggregator_t<NITEMS, GROVE_PER_CLASS_FEW_CLASSES> {
   __device__ __forceinline__ tree_aggregator_t(predict_params params,
                                                void* accumulate_workspace,
                                                void* finalize_workspace)
-    : tmp_storage(finalize_workspace), num_classes(num_classes_),
-      per_thread((vec<NITEMS, float>*)finalize_workspace) {}
+    : num_classes(params.num_classes),
+      per_thread((vec<NITEMS, float>*)finalize_workspace),
+      tmp_storage(params.predict_proba ? per_thread + num_classes
+                                       : finalize_workspace) {}
 
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, float> single_tree_prediction, int tree) {
+    vec<NITEMS, float> single_tree_prediction, int tree, int num_rows) {
+    for (int i = num_rows; i < NITEMS; ++i) acc[i] = 0.0f;
```

Comment:
fixing a bug which somehow did not trigger the test in the previous PR. It triggers it now.